.github/workflows/tests.yml

```diff
@@ -16,32 +16,25 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        target:
+        include:
           - os: linux
             arch: 64bit
+
           - os: mac
             arch: 64bit
+
           - os: windows
             arch: 64bit
-        include:
-          - target:
-              os: linux
-            builder: ubuntu-22.04
-          - target:
-              os: mac
-            builder: macos-12
-          - target:
-              os: windows
-            builder: windows-2022
-
-    name: "${{ matrix.target.os }}-${{ matrix.target.arch }}"
-    runs-on: ${{ matrix.builder }}
+
+    name: "${{ matrix.os }}-${{ matrix.arch }}"
+    runs-on: ${{ matrix.os == 'linux' && 'ubuntu-22.04' ||
+                (matrix.os == 'mac' && 'macos-12' || 'windows-2022') }}
```

Comment:
Why not add an additional `runner` key? I don't really like this approach

Reply:
> We have to ensure that we don't accidentally write some mismatch like:

But that's true of your suggestion solution too.

> When we want to add more jobs to the matrix, there's no single source of truth for the Ubuntu version in every Linux job:

That's true, but I find that unlikely. You could even see it as a feature, where you _can_ deviate if you want to.

> Nitpicking the name, matrix.runner could be confused with the [runner context](https://docs.github.com/en/actions/learn-github-actions/contexts#runner-context)

Valid point. Maybe rename to `runs_on`